apps/main_app/lib/main_app.ex

```diff
@@ -3,19 +3,25 @@ defmodule MainApp do
 
   use Application
 
-  @subscriber Application.compile_env(:main_app, :subscriber, nil)
   @replicator Application.compile_env(:main_app, :replicator, nil)
-  @publisher Application.compile_env(:main_app, :publisher, nil)
+  @publishers Application.compile_env(:main_app, :publishers, [])
 
   @impl true
   def start(_type, _args) do
     children = [
-      @subscriber,
+      MainApp.Communicator,
+      Enum.map(@publishers, fn %{name: name, init_arg: init_arg, module: module} ->
+        %{
+          id: name,
```

Comment:
Maybe if `name` is `nil` we can fallback on `module`? If someone wants to use just one publisher. If he adds more and won't add `name` the `GenServer` will fail if I understand it correctly.

Reply:
That was actually the point. I am planning to add better error checking, but the idea is that if the config is incorrect (e.g. missing `name` the compilation will fail.

I tried defining struct in core and creating it in config, but then I have realized that `config.exs` is executed before the project is compiled (or at least I think so). I was getting struct undefined errors.